src/odemis/driver/test/orsay_test.py

```diff
@@ -0,0 +1,560 @@
+
+
+class TestPneumaticSuspension(unittest.TestCase):
+    """
+    Tests for the pneumatic suspension
+    """
+
+    def setUp(self):
+        """
+        Setup the Orsay client
+        """
+        try:
+            self.oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+        except Exception as e:
+            self.skipTest(e)
+
+        for child in self.oserver.children.value:
+            if child.name == CONFIG_PSUS["name"]:
+                self.psus = child
+
+    def tearDown(self):
+        """
+        Terminate the Orsay client
+        """
+        self.oserver.terminate()
+
+    def test_valve(self):
+        """
+        Test for controlling the power valve
+        """
+        self.psus._valve.Target = 1
+        sleep(0.5)
+        self.assertTrue(self.psus.power.value)
+
+        self.psus._valve.Target = 2
+        sleep(0.5)
+        self.assertFalse(self.psus.power.value)
+
+        self.psus.power.value = True
+        self.assertEqual(self.psus._valve.Target, 1)
+
+        self.psus.power.value = False
+        self.assertEqual(self.psus._valve.Target, 2)
+
+    def test_errorstate(self):
+        """
+        Check that the state VA is updated properly and an exception is raised when the wrong parameter is passed
+        """
+        self.assertRaises(Exception, self.oserver._updateProcessInfo,
+                          {'parameter': self.oserver.datamodel.HybridPlatform.Cancel})
+
+        if not TEST_NOHW:
+            self.skipTest("TEST_NOHW is not set, cannot force data on Actual parameters of Orsay server outside of "
+                          "simulation")
+        test_string = "This thing broke"
+
+        self.oserver.datamodel.HybridPlatform.Manometer2.ErrorState.Actual = test_string
+        self.assertIn("Manometer2", self.psus.state.value)
+        self.assertIn(test_string, self.psus.state.value)
+        self.oserver.datamodel.HybridPlatform.Manometer2.ErrorState.Actual = ""
+
+        self.oserver.datamodel.HybridPlatform.ValvePneumaticSuspension.ErrorState.Actual = test_string
+        self.assertIn("ValvePneumaticSuspension", self.psus.state.value)
+        self.assertIn(test_string, self.psus.state.value)
+        self.oserver.datamodel.HybridPlatform.ValvePneumaticSuspension.ErrorState.Actual = ""
+
+        self.psus._valve.Target = 3
+        sleep(0.5)
+        self.assertIn("ValvePneumaticSuspension is in error", self.psus.state.value)
+        self.psus._valve.Target = -1
+        sleep(0.5)
+        self.assertIn("ValvePneumaticSuspension could not be contacted", self.psus.state.value)
+        self.psus._valve.Target = 1
+
+    def test_updatePower(self):
+        """
+        Check that the power VA is updated correctly and an exception is raised when the wrong parameter is passed
+        """
+        self.assertRaises(Exception, self.psus._updatePower,
+                          {'parameter': self.oserver.datamodel.HybridPlatform.Cancel})
```

Comment:
This type of code is really hard to read. You can write like this:
```
with self.assertRaises(Exception):
    self.psus._updatePower(parameter=self.oserver.datamodel.HybridPlatform.Cancel)
```
Update also the exception type here.

(Please do it everywhere)